libffi's configure assumes that cross-compiled builds are complicated and
introduces convoluted path manipulation involving gcc search paths to the
install paths, resulting in paths like -L/usr/lib/../lib/ appearing in
libffi.pc.  When pkg-config is then used to obtain the linker flags for libffi
it can't tell that this path is on the default search path and returns
$SYSROOT/usr/lib/../lib which then gets written all over the target sysroot.
This then means the sstate can't be shared and triggers QA errors.

As this block is generally pointless, disable it.

Upstream-Status: Inappropriate
Signed-off-by: Ross Burton <ross.burton@intel.com>

Index: libffi-3.2.1/configure.ac
===================================================================
--- libffi-3.2.1.orig/configure.ac
+++ libffi-3.2.1/configure.ac
@@ -592,7 +592,7 @@ AC_ARG_ENABLE(purify-safety,
 
 # These variables are only ever used when we cross-build to X86_WIN32.
 # And we only support this with GCC, so...
-if test "x$GCC" = "xyes"; then
+if false; then
   if test -n "$with_cross_host" &&
      test x"$with_cross_host" != x"no"; then
     toolexecdir="${exec_prefix}"/'$(target_alias)'
